app/assets/javascripts/discourse/app/services/media-optimization-worker.js

```diff
@@ -0,0 +1,130 @@
+import Service from "@ember/service";
+import { getOwner } from "@ember/application";
+import { Promise } from "rsvp";
+import { fileToImageData } from "discourse/lib/media-optimization-utils";
+import { getAbsoluteURL, getURLWithCDN } from "discourse-common/lib/get-url";
+
+export default class MediaOptimizationWorkerService extends Service {
+  appEvents = getOwner(this).lookup("service:app-events");
```

Comment:
Can we not use the inject method for this?

Reply:
There are new methods to use the service in newer versions of EmberJS, but this was the one that worked with a class in 3.15 in my tests. Also the one that is documented at https://guides.emberjs.com/v3.15.0/services/

Reply:
To be clear I'm talking about the method we often use in our codebase already: 

https://github.com/discourse/discourse/blob/master/app/assets/javascripts/admin/addon/components/themes-list.js#L8

That doesn't work in your class?

Reply:
I tried that and it didn't work. Maybe because I'm using native class versus the old Ember `Component.extend({` method?

Reply:
Weird! If you tried that and it didn't work we can leave for now. Eventually we'll get it working.

Reply:
I can clean it up to the newer idiom when we updated Ember!